src/test/java/edu/byu/cs329/cfg/ControlFlowBuilderTests.java

```diff
@@ -1,101 +1,149 @@
+    List<ControlFlowGraph> cfgList = unitUnderTest.build(node);
+    assertEquals(1, cfgList.size());
+    controlFlowGraph = cfgList.get(0);
+    statementTracker = new StatementTracker(node);
+  }
+
+  @Test
+  @Tag("MethodDeclaration")
+  @DisplayName("Should set start and end same when empty method declaration")
+  void should_SetStartAndEndSame_when_EmptyMethodDeclaration() {
+    String fileName = "cfgInputs/should_SetStartAndEndSame_when_EmptyMethodDeclaration.java";
+    init(fileName);
+    assertAll("Method declaration with empty block",
+        () -> assertNotNull(controlFlowGraph.getMethodDeclaration()),
+        () -> assertEquals(controlFlowGraph.getStart(), controlFlowGraph.getEnd())
+    );
+  }
+
+  @Test
+  @Tag("MethodDeclaration")
+  @DisplayName("Should set start to first statement and end different when non-empty method declaration")
+  void should_SetStartToFirstStatementAndEndDifferent_when_NonEmptyMethodDeclaration() {
+    String fileName = "cfgInputs/should_SetStartToFirstStatementAndEndDifferent_when_NonEmptyMethodDeclaration.java";
+    init(fileName);
+    Statement start = controlFlowGraph.getStart();
+    Statement end = controlFlowGraph.getEnd();
+    Statement variableDeclStatement = statementTracker.getVariableDeclarationStatement(0);
+    assertAll("Method declaration with non-empty block",
+        () -> assertNotNull(controlFlowGraph.getMethodDeclaration()), 
+        () -> assertNotEquals(start, end),
+        () -> assertTrue(start == variableDeclStatement),
+        () -> assertTrue(hasEdge(variableDeclStatement, end))
+    );    
+  }
+
+  @Test
+  @Tag("Block")
+  @DisplayName("Should link all when block has no return")
+  void should_LinkAll_when_BlockHasNoReturn() {
+    String fileName = "cfgInputs/should_LinkAll_when_BlockHasNoReturn.java";
+    init(fileName);
+    Statement variableDeclaration = statementTracker.getVariableDeclarationStatement(0);
+    Statement expressionStatement = statementTracker.getExpressionStatement(0);
+    assertTrue(hasEdge(variableDeclaration, expressionStatement));
+  }
+
+  @Test
+  @Tag("Block")
+  @DisplayName("Should link to return when block has return") 
+  void should_LinkToReturn_when_BlockHasReturn() {
+    String fileName = "cfgInputs/should_LinkToReturn_when_BlockHasReturn.java";
+    init(fileName);
+    Statement variableDeclaration = statementTracker.getVariableDeclarationStatement(0);
+    Statement expressionStatement = statementTracker.getExpressionStatement(0);
+    Statement returnStatement = statementTracker.getReturnStatement(0);
+    assertAll(
+        () -> assertTrue(hasEdge(variableDeclaration, returnStatement)),
+        () -> assertFalse(hasEdge(returnStatement, expressionStatement))
+    );
+  }
+
+  // if statement
+  // 1. defined(first(S_Then)) && !isReturn(last(S_Then))
+  // 2. defined(first(S_Then)) && isReturn(last(S_Then))
+  // 3. !defined(first(S_Then))
+  @Test
+  @Tag("IfStatement")

```

Comment:
I like that you tag these as If Statement, it might be nice to make them a nested class as well so that you can see the separation a little easier and run them independently of the other tests if you want.